src/ceph.in

```diff
@@ -299,12 +299,22 @@ def do_extended_help(parser, args):
         if ret:
             print("couldn't get command descriptions for {0}: {1}".\
                 format(target, outs), file=sys.stderr)
+            return 1
         else:
-            help_for_sigs(outbuf.decode('utf-8'), partial)
+            return help_for_sigs(outbuf.decode('utf-8'), partial)
+
+    if filter is True:
+        partial = ' '.join(args)
+    else:
+        partial = None
+
+    if target == None:
+        target = ('mon', '')
+    elif isinstance(target, str) or isinstance(target, unicode):
```

Comment:
it must be None or str now. but we may use it in future and pass a tuple/array. It will be more safe and make this function more generic.